contrib/platform/src/com/sun/jna/platform/win32/Advapi32Util.java

```diff
@@ -275,20 +279,20 @@ public static Account getAccountBySid(String systemName, PSID sid) {
             throw new Win32Exception(rc);
         }
 
-        char[] domainName = new char[cchDomainName.getValue()];
-        char[] name = new char[cchName.getValue()];
+        NativeWin32String domainName = new NativeWin32String(cchDomainName.getValue());
```

Comment:
See remarks for User32.GetWindowText, except this class is where you'll need to switch inside the util function to call the correct function in `Advapi32.java`.  Won't repeat this comment but it applies multiple times.